lib/msf/core/payload/linux/bind_tcp.rb

```diff
@@ -50,20 +46,21 @@ def generate_bind_tcp(opts={})
     Metasm::Shellcode.assemble(Metasm::X86.new, asm).encode_string
   end
 
+  def transport_config(opts={})
+    transport_config_bind_tcp(opts)
+  end
+
   #
   # Determine the maximum amount of space required for the features requested
   #
   def required_space
     # Start with our cached default generated size
-    space = cached_size
+    # TODO: figure out what this should be
+    space = 300
```

Comment:
@hmoore-r7 would appreciate some guidance on how best to approach this stuff. I think I've made a bit of mess here, unsure as to what I should be doing with the payload sizes along the way. The resiliency work that is coming up will address some of this stuff at least, but yeah I can't help but feel like I'm just constantly breaking this bit.